lib/helper.ts

```diff
@@ -241,3 +241,47 @@ export function toLowerKeys(obj: Object): Object {
   }
   return lowerCaseObj;
 }
+
+/**
+ * Constructs a service URL by formatting a parameterized URL.
+ *
+ * @param {string} parameterizedUrl URL that contains variable placeholders, e.g. '{scheme}://ibm.com'.
+ * @param {Map<string, string>} defaultUrlVariables Map from variable names to default values.
+ *  Each variable in the parameterized URL must have a default value specified in this map.
+ * @param {Map<string, string>} providedUrlVariables Map from variable names to desired values.
+ *  If a variable is not provided in this map,
+ *  the default variable value will be used instead.
+ * @returns {string} The formatted URL with all variable placeholders replaced by values.
+ */
+export function constructServiceURL(
```

Comment:
I only just now thought of this, but since we're changing it already... we don't fully capitalize "URL" elsewhere in the core. For example, the BaseService class has a method called `setServiceUrl`. I think we probably want to be consistent and call this `constructServiceUrl`. Sorry for not catching that earlier!

Reply:
I noticed this too, and tbh we have the same discrepancy in the java core.  I only noticed that AFTER the PR was merged in :)
Let's make this change here while we're changing things anyway.

Reply:
BTW, I'm not suggesting we change the Java core :)